### ISSUE TITLE
poetry fails to parse apache-libcloud's python version specification

### ISSUE THREAD

**Issue by deifactor** [python-poetry/poetry#4571]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 11.6
- **Poetry version**: 1.2.0a2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: See below

## Issue

Trying to lock the following `pyproject.toml` fails:

```
[tool.poetry]
name = "badver"
version = "0.1.0"
description = ""
authors = ["Ash <ext0l@catgirl.ai>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.7"
apache-libcloud = "^3.1"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

The error message ultimately ends in

```
  Could not parse version constraint: >=3.5.*

  at ~/Library/Application Support/pypoetry/venv/lib/python3.7/site-packages/poetry/core/semver/helpers.py:140 in parse_single_constraint
      136│         try:
      137│             version = Version.parse(version)
      138│         except ValueError:
      139│             raise ValueError(
    → 140│                 "Could not parse version constraint: {}".format(constraint)
      141│             )
      142│
      143│         if op == "<":
      144│             return VersionRange(max=version)
```

and I think this is because of a line with `    python_requires=">=3.5.*, <4",` in apache-libcloud's setup.py file.

I could try to git them to rephrase their dependencies, but I'd like to be able to install already-existing versions of packages.

This seems related to #3619.

**Comment by deifactor** [comment]
After looking into this, it looks like this is [fixed in poetry-core](https://github.com/python-poetry/poetry-core/commit/4e1f2ab582d1fef0033c0d3f35a3f2f2365a4bc9), but that isn't in the latest version of poetry yet.

**Comment by finswimmer** [comment]
Hello @deifactor,

please don't call it a fix. :) See here why https://github.com/python-poetry/poetry-core/pull/186#issuecomment-903223708

Related discussion: https://github.com/python-poetry/poetry/issues/4176

Yes, the workaround is not included in the current preview release (1.2.0a2).

fin swimmer
